bleak/exc.py

```diff
@@ -16,18 +16,34 @@ class BleakDotNetTaskError(BleakError):
 class BleakDBusError(BleakError):
     """Specialized exception type for D-Bus errors."""
 
-    def __init__(self, dbus_error: str):
+    def __init__(self, dbus_error: str, error_body: list = None):
         """
         Args:
             dbus_error (str): The D-Bus error, e.g. ``org.freedesktop.DBus.Error.UnknownObject``.
+            error_body (str): Optional body of the D-Bus error, sometimes containing error description or details.
         """
-        super().__init__(dbus_error)
+        super().__init__(dbus_error, error_body)
 
     @property
     def dbus_error(self) -> str:
         """Gets the D-Bus error name, e.g. ``org.freedesktop.DBus.Error.UnknownObject``."""
         return self.args[0]
 
+    def __str__(self) -> str:
+        # Error name is always present
+        s = f"[{self.args[0]}]"
+        # Additionally, error body with additional details can be present
+        if len(self.args) > 1 and len(self.args[1]):
```

Comment:
It could be nice to add a property to get `args[1]` as well so that it has a name here.

(this example assumes the unpacking suggestion so that `args[1]` is the same as `error_body[0]`)

```python
@property
def message(self) -> Optional[str]:
    return self.args[1]
```

Reply:
I will add it, however additional check is required, because some errors don't have details and their body is only `[]`, so the `self.args[1]` will fail.